### DISCUSSION TITLE
Font Color Change

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#3451] [Looking for help]
Could you let me know if the font color of the text in the below PDF can be changed to a different color using PyMuPDF and how it can be done?
[Chines.pdf](https://github.com/pymupdf/PyMuPDF/files/15246487/Chines.pdf)

**Comment by JorjMcKie** [comment]
No this is not possible to do.

**Reply by meghanaviyyapu** [reply]
Can you please let me know the reason?

**Reply by JorjMcKie** [reply]
There simply exists no such function.
And it cannot be made reliably.
So you must rewrite the text in the usual manner involving redaction annotations, then looking up the right font, then determining all the properties required for equal looking re-insertion.

**Reply by meghanaviyyapu** [reply]
Can you provide any sample snippet to modify using redaction annotation

**Reply by JorjMcKie** [reply]
[replacer.zip](https://github.com/pymupdf/PyMuPDF/files/15248429/replacer.zip)

**Reply by meghanaviyyapu** [reply]
Thanks a lot

**Reply by meghanaviyyapu** [reply]
Would the approach work for any font type and any language?

**Reply by JorjMcKie** [reply]
I currently see no reason why not ...

**Reply by meghanaviyyapu** [reply]
Have tried the above code to modify color in the PDF but getting error with font. Could you let me know how it can be resolved
![image](https://github.com/pymupdf/PyMuPDF/assets/36301306/b0ac0cc3-d695-46cd-85fb-4b8e8cde70d5)

**Comment by meghanaviyyapu** [comment]
[Chines.pdf](https://github.com/pymupdf/PyMuPDF/files/15271252/Chines.pdf)
This is the PDF for which I tried changing color

**Reply by JorjMcKie** [reply]
Please consult the documentation for the `Font` class. Creating an `Font` object requires either string that represents the name of one of the fonts built into PyMuPDF (like "helv", "tiro" or -if package [pymupdf-fonts](https://pypi.org/project/pymupdf-fonts/) is installed - names like "ubuntu" and more), **_or_** a font name or font buffer (both specified via arguments `fontfile`, respectively `fontbuffer`.
Your argument "MS-Gothic" falls in none of these categories.

**Comment by meghanaviyyapu** [comment]
How to get the fontfile for the fonts that are extracted using span["font"] of PyMuPDF?

**Reply by JorjMcKie** [reply]
look at this example script: [replacer2.zip](https://github.com/pymupdf/PyMuPDF/files/15272938/replacer2.zip)

**Comment by meghanaviyyapu** [comment]
Thanks a lot.

**Reply by meghanaviyyapu** [reply]
Have tried modifying the font color using the attached code but could not view any changes in the text color and could see only rectangles drawn. 
Attached the code and the output PDF. Could you let me know if I missed something?
[color.pdf](https://github.com/pymupdf/PyMuPDF/files/15302669/color.pdf)
[Uploading fontcolos.txt…]()

**Reply by meghanaviyyapu** [reply]
[fontcolos.txt](https://github.com/pymupdf/PyMuPDF/files/15302680/fontcolos.txt)
